keras_nlp/models/opt/opt_tokenizer.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""OPT tokenizer."""
+
+from tensorflow import keras
+
+from keras_nlp.tokenizers.byte_pair_tokenizer import BytePairTokenizer
+
+
+@keras.utils.register_keras_serializable(package="keras_nlp")
+class OPTTokenizer(BytePairTokenizer):
+    """An OPT tokenizer using Byte-Pair Encoding subword segmentation.
+
+    This tokenizer class will tokenize raw strings into integer sequences and
+    is based on `keras_nlp.tokenizers.BytePairTokenizer`. Unlike the
+    underlying tokenizer, it will check for all special tokens needed by OPT
+    models and provides a `from_preset()` method to automatically download
+    a matching vocabulary for a OPT preset.
+
+    This tokenizer does not provide truncation or padding of inputs.
```

Comment:
Is the a BPE thing or OPT thing?

Reply:
This is an all tokenizer thing! This matches our documentation for all tokenizers -> https://keras.io/api/keras_nlp/models/bert/bert_tokenizer/